datetime/src/test/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationUnitParserTest.java

```diff
@@ -0,0 +1,77 @@
+package com.fasterxml.jackson.datatype.jsr310.deser;
+
+import com.fasterxml.jackson.datatype.jsr310.deser.DurationDeserializer.DurationUnitParser;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+import static java.util.Optional.of;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class DurationUnitParserTest {
+
+    private final String stringPattern;
+    private final TemporalUnit temporalUnit;
+
+    public DurationUnitParserTest(String stringPattern, TemporalUnit temporalUnit) {
+        this.stringPattern = stringPattern;
+        this.temporalUnit = temporalUnit;
+    }
+
+    @Test
+    public void shouldMapToTemporalUnit() {
+        Optional<DurationUnitParser> durationPattern = DurationUnitParser.from(stringPattern);
+
+        assertEquals(of(temporalUnit), durationPattern.map(dp -> dp.unit));
+    }
+
+    @Parameters
+    public static Collection<Object[]> testCases() {
+        List<Object[]> baseTestCases = asList(
+                asArray("Nanos", ChronoUnit.NANOS),
+                asArray("Micros", ChronoUnit.MICROS),
+                asArray("Millis", ChronoUnit.MILLIS),
+                asArray("Seconds", ChronoUnit.SECONDS),
+                asArray("Minutes", ChronoUnit.MINUTES),
+                asArray("Hours", ChronoUnit.HOURS),
+                asArray("HalfDays", ChronoUnit.HALF_DAYS),
+                asArray("Days", ChronoUnit.DAYS),
+                asArray("Weeks", ChronoUnit.WEEKS),
+                asArray("Months", ChronoUnit.MONTHS),
+                asArray("Years", ChronoUnit.YEARS),
+                asArray("Decades", ChronoUnit.DECADES),
+                asArray("Centuries", ChronoUnit.CENTURIES),
+                asArray("Millennia", ChronoUnit.MILLENNIA),
+                asArray("Eras", ChronoUnit.ERAS),
+                asArray("Forever", ChronoUnit.FOREVER)
```

Comment:
TIL: There is a `ChronoUnit` for "forever". Wondering if we can trim down the actual realistically supported `ChronoUnit`s?

Reply:
Not supported, fixed. Actually everthing greater than `DAYS` is not supported because is considered not precise. I fixed the code and added some tests.